heroku/cli
# chore: add spell check

### Description
[Work Item](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001teOSEYA2/view)

This pull request adds a job to the Tests workflow and a Git pre-commit hook to the CLI repository, both dedicated to spell checking modified files. The spell checker will generate a list of any misspelled words it detects, including the file and line number. Also, there's a `custom-dictionary.txt` file that should be updated with words, variables, package names, etc. that are flagged by the spell checker but are not actually misspelled. 

The `spell-check` job in the Tests workflow will fail if any spelling errors are detected, and the Git pre-commit hook will block the commit. To bypass the Git pre-commit hook and commit changes without running the spell check, use the `-n` flag.

### Testing
1. create a new branch based on this branch
2. run `yarn`
3. create a test file and add misspellings within it
4. attempt to commit your changes
NOTE: There should be a Git error when you try to commit, that's the expected behavior of the Git pre-commit hook due to the misspellings.
5. run `git commit` with the `-n` flag and push changes to the branch
NOTE: This will bypass the Git pre-commit hook and initiate the Tests workflow, which should also fail as expected. 
6. correct the misspellings in the test file
7. commit and push to the branch (without the `-n` flag)
NOTE: At this point, the commit should go through and the Tests workflow should pass.
